commands/chargen.py

```diff
@@ -0,0 +1,190 @@
+                    self.session.execute_cmd('@charcreate {}'.format(
+                        new_character.key
+                    ))
+                    return
+            else:
+                self.msg("That is not a valid character choice.")
+                return
+        super(CmdIC, self).func()
+
+
+class CmdCharCreate(MuxPlayerCommand):
+    """
+    create a new character
+
+    Usage:
+      @charcreate <charname>
+
+    Create a new character. You may use upper-case letters in the
+    name - you will nevertheless always be able to access your
+    character using lower-case letters if you want.
+    """
+    key = "@charcreate"
+    locks = "cmd:pperm(Players)"
+    help_category = "General"
+
+    def func(self):
+        "create the new character"
+        player = self.player
+        session = self.session
+        if not self.args:
+            self.msg("Usage: @charcreate <charname>")
+            return
+        key = self.args.strip()
+
+        charmax = _MAX_NR_CHARACTERS if _MULTISESSION_MODE > 1 else 1
+
+        if not player.is_superuser and \
+            (player.db._playable_characters and
+                len(player.db._playable_characters) >= charmax):
+            self.msg("You may only create a maximum of %i characters." % charmax)
+            return
+
+        # create the character
+        from evennia.objects.models import ObjectDB
+
+        start_location = ObjectDB.objects.get_id(settings.START_LOCATION)
+        default_home = ObjectDB.objects.get_id(settings.DEFAULT_HOME)
+        typeclass = settings.BASE_CHARACTER_TYPECLASS
+        permissions = settings.PERMISSION_PLAYER_DEFAULT
+
+        # check whether a character already exists
+        new_character = None
+        candidates = search.objects(key, typeclass='typeclasses.characters.Character')
+        if candidates:
+            for c in candidates:
+                if c.access(player, 'puppet'):
+                    new_character = c
+                    break
+
+        startnode = "menunode_welcome_archetypes"
+        if not new_character:
+
+            new_character = create.create_object(typeclass, key=key,
+                                                 location=None,
+                                                 home=default_home,
+                                                 permissions=permissions)
+            # only allow creator (and immortals) to puppet this char
+            new_character.locks.add("puppet:id(%i) or pid(%i) or perm(Immortals) or pperm(Immortals)" %
+                                    (new_character.id, player.id))
+            player.db._playable_characters.append(new_character)
+
+        else:
+            if new_character.db.chargen_complete:
+                self.msg(("{name} has already completed character "
+                          "generation. Use @ic {name} to puppet.").format(
+                            name=new_character.key
```

Comment:
I've consistently used `format` in the code I've written. I began with the original CmdCharCreate code, which used `%`. I'll go ahead and change them.